examples/llm/inference/README.md

```diff
@@ -0,0 +1,136 @@
+# LLM Inference
+
+This folder contains helper scripts for exporting and generating outputs with  your Composer-trained LLMs.
+
+
+## Converting a Composer checkpoint to a HF checkpoint folder
+
+The LLMs trained with this codebase are all HuggingFace (HF) PretrainedModels, which we wrap with a `HuggingFaceModel` wrapper class to make compatible with Composer. See details [here](https://docs.mosaicml.com/en/latest/examples/pretrain_finetune_huggingface.html).
+
+At the end of your training runs, you will see a collection of Composer trainer checkpoints such as `ep0-ba2000-rank0.pt`. These checkpoints contain the entire training state, including the model, tokenizer, optimizer state, schedulers, timestamp, metrics, etc. Though these Composer checkpoints are useful during training, at inference time we usually just want the model, tokenizer, and metadata.
+
+To extract these pieces, we provide a script `convert_composer_to_hf.py` that converts a Composer checkpoint directly to a standard HF checkpoint folder. For example:
+```bash
+python convert_composer_to_hf.py --composer_path ep0-ba2000-rank0.pt --hf_output_path my_hf_model/ --output_precision bf16
+```
+
+This will produce a folder like:
+```
+my_hf_model/
+  config.json
+  merges.txt
+  pytorch_model.bin
+  special_tokens_map.json
+  tokenizer.json
+  tokenizer_config.json
+  vocab.json
+```
+which can be loaded with standard HF utilities like `AutoModelForCausalLM.from_pretrained('my_hf_model')`.
+
+You can also pass object store URIs for both `--composer_path` and `--hf_output_path` to easily convert checkpoints stored in S3, OCI, etc.
+
+### IMPORTANT NOTE:
+If you trained and saved a custom HF model such as `MosaicGPT`, then in any external inference codebase, you need to import and register the new model class and config before auto classes like `AutoModel` will work. For example:
+
+```python
+# MosaicGPT, MosaicGPTConfig source code live in this repo
+# pip install mosaicml-examples
```

Comment:
```suggestion
# pip install <my-awesome-repo>
```

Reply:
otherwise it looks like there is a pip package called `mosaicml-examples`, which there is not (yet)